src/WebJobs.Script.WebHost/Controllers/HostController.cs

```diff
@@ -154,7 +163,57 @@ public async Task<IActionResult> SyncTriggers()
         public IActionResult Restart([FromServices] IScriptHostManager hostManager)
         {
             Task ignore = hostManager.RestartHostAsync();
-            return Ok(_webHostSettings.Value);
+            return Ok(_applicationHostOptions.Value);
+        }
+
+        /// <summary>
+        /// Currently this endpoint only supports taking the host offline and bringing it back online.
+        /// </summary>
+        /// <param name="state">The desired host state. See <see cref="ScriptHostState"/>.</param>
+        [HttpPut]
+        [Route("admin/host/state")]
+        [Authorize(Policy = PolicyNames.AdminAuthLevel)]
+        public async Task<IActionResult> SetState()
+        {
+            if (string.Compare(Request.ContentType, "text/plain", StringComparison.OrdinalIgnoreCase) != 0)
+            {
+                return BadRequest();
+            }
+
+            string state;
+            using (StreamReader sr = new StreamReader(Request.Body, Encoding.UTF8))
```

Comment:
ASP.NET doesn't allow binding to a simple string value via param so I have to read this manually

Reply:
feels a bit weird that this a text plain API and it results in this awkward reading. Why not:
```
{
 "status": "offline"
}
``` 
or something similar?

Reply:
or just `"offline"` with content type application/json

Reply:
Fixed